.github/workflows/handler-pr-close.yml

```diff
@@ -0,0 +1,134 @@
+        env:
+          GOOGLE_CREDENTIALS: ${{ secrets.GOOGLE_CREDENTIALS }}
+          GOOGLE_PROJECT: ${{ secrets.GOOGLE_PROJECT }}
+          GOOGLE_REGION: ${{ secrets.GOOGLE_REGION }}
+          GOOGLE_ZONE: ${{ secrets.GOOGLE_ZONE }}
+          GOOGLE_SERVICE_ACCOUNT: ${{ secrets.GOOGLE_SERVICE_ACCOUNT }}
+          TFE_HOSTNAME: ${{ secrets.TFE_HOSTNAME }}
+          TFE_ORGANIZATION: ${{ secrets.TFE_ORGANIZATION }}
+          TFE_TOKEN: ${{ secrets.TFE_TOKEN }}
+          TFE_WORKSPACE: ${{ secrets.TFE_WORKSPACE }}
+        run: |
+          cat <<EOF > github.auto.tfvars
+          google = {
+            credentials     = <<-EOC
+            $GOOGLE_CREDENTIALS
+            EOC
+            project         = "$GOOGLE_PROJECT"
+            region          = "$GOOGLE_REGION"
+            zone            = "$GOOGLE_ZONE"
+            service_account = "$GOOGLE_SERVICE_ACCOUNT"
+          }
+          tfe = {
+            hostname     = "$TFE_HOSTNAME"
+            organization = "$TFE_ORGANIZATION"
+            token        = "$TFE_TOKEN"
+            workspace    = "$TFE_WORKSPACE"
+          }
+          EOF
+    # this step will get the workspace created for google-private-tcp-active-active for this PR
+      - name: Get workspace for google-private-tcp-active-active
+        id: get-workspaces-google-private-tcp-active-active
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        run: >-
+          echo "::set-output name=WORSPACE_TCP_AA::$(terraform workspace list -input=false -no-color | grep google-private-tcp-active-active-${{ github.event.client_payload.github.payload.issue.number }})"
+
+          echo "::set-output name=WORSPACE_TCP_AA_EXISTS::$(terraform workspace list -input=false -no-color | grep -cx google-private-tcp-active-active-${{ github.event.client_payload.github.payload.issue.number }})"
+    # this step will delete resources if workspace exists for this PR
+      - name: Terraform Destroy
+        id: destroy
+        if: github.event.pull_request.merged == true && steps.get-workspaces-google-private-tcp-active-active.outputs.WORSPACE_TCP_AA_EXISTS == "1"
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        run: terraform destroy -auto-approve -input=false -no-color
+    # this step will delete workspace for this PR
+      - name: Terraform Delete Workspace
+        id: destroy-workspace
+        if: github.event.pull_request.merged == true && steps.get-workspaces-google-private-tcp-active-active.outputs.WORSPACE_TCP_AA_EXISTS == "1" && steps.destroy.outcome == 'success'
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        env:
+          TFE_HOSTNAME: ${{ secrets.TFE_HOSTNAME }}
+          TFE_TOKEN: ${{ secrets.TFE_TOKEN }}
+        run: terraform workspace delete "${{ steps.get-workspaces-google-private-tcp-active-active.outputs.WORSPACE_TCP_AA }}" -no-color
+     # this step will delete workspace for this PR
+      - name: Update comment
```

Comment:
I'm wondering if the comment update logic here makes sense.  I think the event payload from github won't have a comment id because the initiating event is the PR closing.  You might be able to add a new comment to the PR though

Reply:
alternatively I don't think we have any explicit need to comment anything here when the workspace is destroyed.  You could just nuke this phase out of the workflow and we can add notification later if needed

Reply:
removed